R/rdeephaven/R/update_by_ops_wrapper.R

```diff
@@ -0,0 +1,615 @@
+  return(UpdateByOp$new(INTERNAL_emstd_time(ts_col, decay_time, cols, operation_control$.internal_rcpp_object)))
+}
+
+#' @description
+#' Creates a rolling sum UpdateByOp for each column in `cols`, using ticks as the windowing unit.
+#' Ticks are row counts, and you may specify the reverse and forward window in number of rows to include.
+#' The current row is considered to belong to the reverse window but not the forward window.
+#' Also, negative values are allowed and can be used to generate completely forward or completely reverse windows.
+#' @param cols String or list of strings denoting the column(s) to operate on. Can be renaming expressions, i.e. “new_col = col”.
+#' Default is to compute the rolling sum for all non-grouping columns.
+#' @param rev_ticks Integer scalar denoting the look-behind window size in number of rows.
+#' @param fwd_ticks Integer scalar denoting the look-ahead window size in number of rows. Default is 0.
+#' @export
+udb_roll_sum_tick <- function(cols, rev_ticks, fwd_ticks = 0) {
+  verify_string("cols", cols, FALSE)
+  verify_any_int("rev_ticks", rev_ticks, TRUE)
+  verify_any_int("fwd_ticks", fwd_ticks, TRUE)
+  return(UpdateByOp$new(INTERNAL_rolling_sum_tick(cols, rev_ticks, fwd_ticks)))
+}
+
+#' @description
+#' Creates a rolling sum UpdateByOp for each column in `cols`, using time as the windowing unit.
+#' This uses ISO-8601 time strings as the reverse and forward window parameters.
+#' Negative values are allowed and can be used to generate completely forward or completely reverse windows.
+#' A row containing a null value in the timestamp column belongs to no window and will not be considered
+#' in the windows of other rows; its output will be null.
+#' @param ts_col String denoting the column to use as the timestamp.
+#' @param cols String or list of strings denoting the column(s) to operate on. Can be renaming expressions, i.e. “new_col = col”.
+#' Default is to compute the rolling sum for all non-grouping columns.
+#' @param rev_time ISO-8601-formatted string specifying the look-behind window size.
+#' @param fwd_time ISO-8601-formatted string specifying the look-ahead window size. Default is 0 seconds.
+#' @export
+udb_roll_sum_time <- function(ts_col, cols, rev_time, fwd_time = "PT0s") {
+  verify_string("ts_col", ts_col, TRUE)
+  verify_string("cols", cols, FALSE)
+  verify_string("rev_time", rev_time, TRUE)
+  verify_string("fwd_time", fwd_time, TRUE)
+  return(UpdateByOp$new(INTERNAL_rolling_sum_time(ts_col, cols, rev_time, fwd_time)))
+}
+
+#' @description
+#' Creates a rolling group UpdateByOp for each column in `cols`, using ticks as the windowing unit.
+#' Ticks are row counts, and you may specify the reverse and forward window in number of rows to include.
+#' The current row is considered to belong to the reverse window but not the forward window.
+#' Also, negative values are allowed and can be used to generate completely forward or completely reverse windows.
```

Comment:
In python, all of our windowed operations have detailed examples for defining the windows.  I think we should have something similar in R, unless there is a major, compelling reason not to.  We did this because it was too confusing without examples.

e.g.
https://deephaven.io/core/pydoc/code/deephaven.updateby.html#deephaven.updateby.rolling_avg_tick
https://deephaven.io/core/pydoc/code/deephaven.updateby.html#deephaven.updateby.rolling_avg_time

Reply:
I think there is a good reason not to do this. These docstrings are not the literal characters that are going to get turned into R HTML docs. I am going to split the within-R-studio docs into reference-style pages and tutorial-style pages. Most within-R-studio documentation sort of shoves both of these things into one doc, but I think our stuff is sufficiently complicated to warrant not doing that. In that case, the tutorial-style doc will have the windowing examples you speak of. All R docs should also have actual examples demonstrating usage of the thing they're documenting, so we will have that as well. I think there is not a need to clutter these docstrings with examples, as they are really only intended to be read from source.

Reply:
It's worth mentioning that the doc split I'm proposing breaks R convention and breaks the typical roxygen workflow. I'm ok with this, as both of those things are bad.

Reply:
This also inclines me to shorten these window docs even further, to something like 
```
Creates a rolling group UpdateByOp for each column in 'cols', using ticks as the windowing unit.
Ticks are row counts, and you may specify the reverse and forward window in number of rows to include.
```

Reply:
No part of this plan makes sense to me.

Reply:
In particular, users almost never read all of the documentation.  They need what they want at the point they ask for it.  By removing the documentation from `?`, they will never see what they need to.